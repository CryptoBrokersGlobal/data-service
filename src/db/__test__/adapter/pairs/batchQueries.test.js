const { identity, map, path } = require('ramda');

const batchQueries = require('../../../adapter/pairs/batchQueries');
const sql = require('../../../sql');

const tMock = {
  batch: identity,
  oneOrNone: identity,
};

const spiedSql = map(jest.fn, sql.build.pairs);
afterEach(() => {
  jest.clearAllMocks();
});

describe('Pairs createDbTask function', () => {
  const getCalls = spiesObj => map(path(['mock', 'calls']), spiesObj);

  it('covers case when WAVES — amount asset', () => {
    const pair = {
      amountAsset: 'WAVES',
      priceAsset: 'Ft8X1v1LTa1ABafufpaCWyVj8KkaxUWE6xBhW6sNFJck',
    };
    batchQueries(spiedSql, pair, tMock);
    expect(getCalls(spiedSql)).toMatchSnapshot();
  });

  it('covers case when WAVES — price asset', () => {
    const pair = {
      amountAsset: '474jTeYx2r2Va35794tCScAXWJG9hU2HcgxzMowaZUnu',
      priceAsset: 'WAVES',
    };
    batchQueries(spiedSql, pair, tMock);
    expect(getCalls(spiedSql)).toMatchSnapshot();
  });

  it('covers case when WAVES is neither price nor amount asset', () => {
    const pair = {
      amountAsset: '474jTeYx2r2Va35794tCScAXWJG9hU2HcgxzMowaZUnu',
      priceAsset: 'Ft8X1v1LTa1ABafufpaCWyVj8KkaxUWE6xBhW6sNFJck',
    };
    batchQueries(spiedSql, pair, tMock);
    expect(getCalls(spiedSql)).toMatchSnapshot();
  });
});
