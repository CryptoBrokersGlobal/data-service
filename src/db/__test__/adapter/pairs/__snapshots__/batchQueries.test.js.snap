// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pairs createDbTask function covers case when WAVES is neither price nor amount asset 1`] = `
Object {
  "decimals": Array [
    Array [
      "474jTeYx2r2Va35794tCScAXWJG9hU2HcgxzMowaZUnu",
    ],
    Array [
      "Ft8X1v1LTa1ABafufpaCWyVj8KkaxUWE6xBhW6sNFJck",
    ],
  ],
  "firstPrice": Array [
    Array [
      Object {
        "amountAsset": "474jTeYx2r2Va35794tCScAXWJG9hU2HcgxzMowaZUnu",
        "priceAsset": "Ft8X1v1LTa1ABafufpaCWyVj8KkaxUWE6xBhW6sNFJck",
      },
    ],
  ],
  "lastPrice": Array [
    Array [
      Object {
        "amountAsset": "474jTeYx2r2Va35794tCScAXWJG9hU2HcgxzMowaZUnu",
        "priceAsset": "Ft8X1v1LTa1ABafufpaCWyVj8KkaxUWE6xBhW6sNFJck",
      },
    ],
  ],
  "volume": Array [
    Array [
      Object {
        "amountAsset": "474jTeYx2r2Va35794tCScAXWJG9hU2HcgxzMowaZUnu",
        "priceAsset": "Ft8X1v1LTa1ABafufpaCWyVj8KkaxUWE6xBhW6sNFJck",
      },
    ],
  ],
  "volumeInPriceAsset": Array [
    Array [
      Object {
        "amountAsset": "474jTeYx2r2Va35794tCScAXWJG9hU2HcgxzMowaZUnu",
        "priceAsset": "Ft8X1v1LTa1ABafufpaCWyVj8KkaxUWE6xBhW6sNFJck",
      },
    ],
  ],
  "weightedAveragePrice": Array [
    Array [
      Object {
        "asset1": "Ft8X1v1LTa1ABafufpaCWyVj8KkaxUWE6xBhW6sNFJck",
        "asset2": "WAVES",
      },
    ],
  ],
}
`;

exports[`Pairs createDbTask function covers case when WAVES — amount asset 1`] = `
Object {
  "decimals": Array [
    Array [
      "WAVES",
    ],
    Array [
      "Ft8X1v1LTa1ABafufpaCWyVj8KkaxUWE6xBhW6sNFJck",
    ],
  ],
  "firstPrice": Array [
    Array [
      Object {
        "amountAsset": "WAVES",
        "priceAsset": "Ft8X1v1LTa1ABafufpaCWyVj8KkaxUWE6xBhW6sNFJck",
      },
    ],
  ],
  "lastPrice": Array [
    Array [
      Object {
        "amountAsset": "WAVES",
        "priceAsset": "Ft8X1v1LTa1ABafufpaCWyVj8KkaxUWE6xBhW6sNFJck",
      },
    ],
  ],
  "volume": Array [
    Array [
      Object {
        "amountAsset": "WAVES",
        "priceAsset": "Ft8X1v1LTa1ABafufpaCWyVj8KkaxUWE6xBhW6sNFJck",
      },
    ],
  ],
  "volumeInPriceAsset": Array [],
  "weightedAveragePrice": Array [],
}
`;

exports[`Pairs createDbTask function covers case when WAVES — price asset 1`] = `
Object {
  "decimals": Array [
    Array [
      "474jTeYx2r2Va35794tCScAXWJG9hU2HcgxzMowaZUnu",
    ],
    Array [
      "WAVES",
    ],
  ],
  "firstPrice": Array [
    Array [
      Object {
        "amountAsset": "474jTeYx2r2Va35794tCScAXWJG9hU2HcgxzMowaZUnu",
        "priceAsset": "WAVES",
      },
    ],
  ],
  "lastPrice": Array [
    Array [
      Object {
        "amountAsset": "474jTeYx2r2Va35794tCScAXWJG9hU2HcgxzMowaZUnu",
        "priceAsset": "WAVES",
      },
    ],
  ],
  "volume": Array [
    Array [
      Object {
        "amountAsset": "474jTeYx2r2Va35794tCScAXWJG9hU2HcgxzMowaZUnu",
        "priceAsset": "WAVES",
      },
    ],
  ],
  "volumeInPriceAsset": Array [
    Array [
      Object {
        "amountAsset": "474jTeYx2r2Va35794tCScAXWJG9hU2HcgxzMowaZUnu",
        "priceAsset": "WAVES",
      },
    ],
  ],
  "weightedAveragePrice": Array [],
}
`;
